### ISSUE TITLE
implicit_assignment_linter misses scoped usage with '&&'

### ISSUE THREAD

**Issue by MichaelChirico** [jimhester/lintr#2918]
I think Tim's case was addressed so feel free to merge.

I think we have however another edge case to cover at some point:

``` r
lintr::lint(
  text = "if (a <- 1 && b <- 2) { print(a) }
  print(a)", 
  linters = lintr::implicit_assignment_linter(allow_scoped = TRUE)
)
#> <text>:1:5: warning: [implicit_assignment_linter] Avoid implicit assignments in function calls. For example, instead of `if (x <- 1L) { ... }`, write `x <- 1L; if (x) { ... }`.
#> if (a <- 1 && b <- 2) { print(a) }
#>     ^~~~~~~~~~~~~~~~

lintr::lint(
  text = "if (a <- 1 && b <- 2) { print(b) }
  print(b)", 
  linters = lintr::implicit_assignment_linter(allow_scoped = TRUE)
)
#> ℹ No lints found.
```

<sup>Created on 2025-08-03 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

_Originally posted by @Bisaloo in https://github.com/r-lib/lintr/pull/2914#pullrequestreview-3081856998_
